lumicks/pylake/calibration.py

```diff
@@ -1,3 +1,218 @@
+
+            # Slice the data corresponding to the calibration we want to reproduce.
+            calib_slice = f.force1x[calibration]
+
+            # De-calibrate to volts using the calibration that was active before this slice.
+            previous_calibration = calib_slice.calibration[0]
+            calib_slice = calib_slice / previous_calibration.force_sensitivity
+
+            calibration_params = previous_calibration.calibration_params()
+            new_calibration = lk.calibrate_force(calib_slice.data, **calibration_params)
+            new_calibration.plot()
+
+            # Make a new calibration, but change the amount of blocking
+            less_blocking_params = calibration_params | {"num_points_per_block": 200}
+            less_blocking = lk.calibrate_force(calib_slice.data, **less_blocking_params)
+            less_blocking.plot()
+        """
+        model_params = self._model_params()
+        model_params["active_calibration"] = self.active_calibration
+
+        def check_defined(pair):
+            return pair[1] is not None
+
+        return dict(filter(check_defined, (self.power_spectrum_params() | model_params).items()))
+
+    @property
+    def excluded_ranges(self):
+        """Returns the frequency exclusion ranges
+
+        When fitting power spectra, specific frequency intervals of the spectrum can be ignored to
+        exclude noise peaks. This attribute returns a list of these intervals in Hertz."""
+        self._verify_full()
+
+        exclusion_range_indices = [
+            int(match[0])
+            for key in self.data.keys()
+            if (match := re.findall(r"Exclusion range ([\d+]) \(min\.\) \(Hz\)", key))
+        ]
+        return [
+            (
+                self.data[f"Exclusion range {exclusion_idx} (min.) (Hz)"],
+                self.data[f"Exclusion range {exclusion_idx} (max.) (Hz)"],
+            )
+            for exclusion_idx in sorted(exclusion_range_indices)
+        ]
+
+    @property
+    def fit_range(self):
+        """Returns the spectral fit range used for the calibration"""
+        self._verify_full()
+
+        return (
+            self.data["Fit range (min.) (Hz)"],
+            self.data["Fit range (max.) (Hz)"],
+        )
+
+    @property
+    def sample_rate(self):
+        """Returns the data sample rate"""
+        self._verify_full()
+
+        return self.data.get("Sample rate (Hz)")
+
+    @property
+    def active_calibration(self):
+        """Returns whether it was an active calibration or not"""
+        self._verify_full()
+
+        return self.data.get("driving_frequency (Hz)") is not None
+
+    @property
+    def fast_sensor(self):
+        """Returns whether it was a fast sensor or not"""
+        self._verify_full()
+
+        # If it is not a fixed diode or free diode, it's a fast sensor
+        diode_fields = ("Diode frequency (Hz)", "f_diode (Hz)")
+        return not any(f in self.data.keys() for f in diode_fields)
```

Comment:
I would just put the tuple directly in the generator

```suggestion
        return not any(f in self.data.keys() for f in ("Diode frequency (Hz)", "f_diode (Hz)"))
```